README.md

```diff
@@ -11,25 +9,71 @@ TESTING
+The library is available from [PyPI](#TODO) and can be installed with pip:
 
 > pip install -U harmony-py
 
+This will install harmony-py and its dependencies into your current Python environment. It's recommended that you install harmony-py into a virtualenv along with any other dependencies you may have.
 
-## Developing
 
-Install dependencies:
+# Running Examples & Developing on Harmony Py
+
+## Prerequisites
+
+* Python 3.7+
+* (optional,recommended) [pyenv](https://github.com/pyenv/pyenv) and [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv)
+
+
+## Installing Development & Example Dependencies
+
+First, it's recommended that you create a Python virtualenv so that Harmony Py and its dependencies are isolated in their own environment. To do so, you can either [create and activate a Python virtual environment with venv](https://docs.python.org/3/tutorial/venv.html), or--if you have pyenv and pyenv-virtualenv installed--use pyenv to create and activate one for you (`harmony-py`). There are `make` targets for both of these options--choose one.
+
+1a. Create a virtualenv with venv:
+
+> make venv-setup
+> source .venv/bin/activate
+
+To deactivate it:
+> source deactivate
+
+1b. Use pyenv & pyenv-virtualenv. This will install Python 3.9 & create a virtualenv using that version of Python. Important: if this is your first time using pyenv to install Python, be sure that you have the [Python build requirements installed](https://github.com/pyenv/pyenv/wiki#suggested-build-environment) first.
+
+> make pyenv-setup
+
+If you've setup pyenv with your shell properly, it should automatically activate the environment. It does this by creating a `.python-version` file in the project directory. Most shells can be setup to automatically activate & deactivate virtual environments when cd'ing into & out of directories by using the value found in `.python-version`. This is convenient since it ensures that the correct virtualenv has been activated (and deactivated) when starting work on a project. See the pyenv docs for more details. If you need to manually activate & deactivate:
+
+> pyenv activate harmony-py
+> pyenv deactivate
+
+2. Install dependencies:
 
 > make install
 
-Optionally register your local copy with pip:
+3. Optionally register your local copy with pip:
 
 > pip install -e ./path/to/harmony_py
 
 
+## Running the Example Jupyter Notebooks
+
+Jupyter notebooks in the `examples` subdirectory show how to use the Harmony Py library. Start up the Jupyter Lab notebook server and run these examples: 
+
+The Jupyter Lab server will start and [open in your browser](http://localhost:8888/lab). Double-click on a notebook in the file-browser sidebar and run the notebook. Note that some notebooks may have cells which prompt for your EDL username and password. Be sure to use your UAT credentials since all of the example notebooks use the Harmony UAT environment.
+
+> make examples
```

Comment:
Noting that I received this error with option 1b above, but option 1a worked for me:

```ERROR: Cannot uninstall 'terminado'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.```

Reply:
iirc, this was due to pyenv reporting it wasn't installed in the shell?

Reply:
I don't think we ever concluded that the shell install issue was the cause of the `terminado` error but I certainly could have missed that. I know we discovered the `make venv-setup` approached was successful so we were able to  move on.

Reply:
Okay, thanks. The virtualenv sol'n is fine, but I'm interested to see if someone with a Mac who uses pyenv can help resolve this.